covid19/spreadsheet_validator.py

```diff
@@ -0,0 +1,100 @@
+from rows.fields import IntegerField
+
+from django.forms import ValidationError
+
+from brazil_data.cities import get_city_info, get_state_info
+
+
+TOTAL_LINE_DISPLAY = 'TOTAL NO ESTADO'
+UNDEFINED_DISPLAY = 'Importados/Indefinidos'
+INVALID_CITY_CODE = -1
+
+
+def format_spreadsheet_rows_as_dict(rows_table, date, state):
+    """
+    Receives rows.Table object, a date and a brazilan UF, validates the data
+    and returns a dict in the formatted data:
+
+    This is an auxiliary method used by covid19.forms.StateSpreadsheetForm with the uploaded file
+    """
+    # TODO: https://github.com/turicas/brasil.io/issues/210
+    field_names = rows_table.field_names
+
+    confirmed_attr = _get_column_name(
+        field_names, ['confirmados', 'confirmado', 'casos_confirmados']
+    )
+    deaths_attr = _get_column_name(field_names, ['obitos', 'obito', 'morte', 'mortes'])
+    city_attr = _get_column_name(field_names, ['municipio', 'cidade'])
+
+    if not rows_table.fields[confirmed_attr] == IntegerField:
+        raise ValidationError('A coluna "Confirmados" precisa ter somente números inteiros"')
+    if not rows_table.fields[deaths_attr] == IntegerField:
+        raise ValidationError('A coluna "Mortes" precisa ter somente números inteiros"')
+
+    results = []
+    bad_cities = []
+    has_total, has_undefined = False, False
+    for i, entry in enumerate(rows_table):
+        city = getattr(entry, city_attr, None)
+        confirmed = getattr(entry, confirmed_attr, None)
+        deaths = getattr(entry, deaths_attr, None)
+
+        if confirmed is None:
```

Comment:
Essa verificação está incorreta. Se a pessoa baixar a planilha-modelo e não preencher um dado para um município (porque não há casos), então isso não deve gerar um erro. O erro deve acontecer se `not confirmed and deaths or not deaths and confirmed`.

Reply:
Resolvido em 75eedb2